great_expectations/render/renderer/site_builder.py

```diff
@@ -971,6 +977,25 @@ def _add_validations_to_index_links(
                     logger.warning(error_msg)
 
 
+def _resolve_asset_name(validation_results: ExpectationValidationResult) -> str | None:
+    """
+    Resolve the asset name from the validation results meta data.
+    FDS does not store data_asset_name in batch_kwargs or batch_spec and it must be
+    pulled from the active batch definition.
+    """
```

Comment:
@morphatic Thanks so much for the detailed issue description.
I ended up using exactly what you suggested.

Would you like me to make you a CoAuthor on this PR?

Reply:
Glad it helped! I would not complain about being listed as a co-author. Thank you for asking!